topo_processor/stac/validate_report.py

```diff
@@ -0,0 +1,23 @@
+from typing import Dict, List
+
+
+class ValidateReport:
+    total: int
+    report_per_error_type: Dict[str, Dict[str, int]]
+
+    def __init__(self):
+        self.total = 0
+        self.report_per_error_type = {}
+
+    def add_errors(self, errors_per_schema: Dict[str, List[str]]) -> None:
+        if errors_per_schema:
+            for schema_uri in errors_per_schema:
+                if schema_uri not in self.report_per_error_type:
```

Comment:
this is doing a ton of dict lookups, a common pattern to reduce the lookups

```python
existing = dict.get(schema_uri)
if existing is None:
   existing = {}
   dict[schema_uri] = existing
```

For incrementing Id recommend something like

```
existing[error] = existing.get(error, 0) + 1
```

Reply:
Just doubled checked the performance of the two

| Command | Mean [ms] | Min [ms] | Max [ms] | Relative |
|:---|---:|---:|---:|---:|
| `dict_get` | 283.9 ± 5.1 | 275.7 | 291.4 | 1.00 |
| `dict_array` | 349.2 ± 6.8 | 337.6 | 359.1 | 1.23 ± 0.03 |

```python
    def dict_array(self, a, b) -> None:
        if not a in self.dict:
            self.dict[a] = {}
        if not b in self.dict[a]:
            self.dict[a][b] = 1
        else:
            self.dict[a][b] = self.dict[a][b] + 1

    def dict_get(self, a, b) -> None:
        existing = self.dict.get(a)
        if existing is None:
            self.dict[a] = existing = {}

        existing[b] = existing.get(b, 0) + 1
```

I do wonder if there is a more pythonic way of doing

```python
existing = self.dict.get(a)
if existing is None:
    self.dict[a] = existing = {}
```